File: MVVMFrame/src/main/java/com/leonyr/mvvm/net/cookie/CookieManger.java
package com.leonyr.mvvm.net.cookie;

import android.content.Context;

import java.util.List;

import okhttp3.Cookie;
import okhttp3.CookieJar;
import okhttp3.HttpUrl;

/**
 * ==============================================================
 * Description:
 * <p>
 * Created by 01385127 on 2019.04.29
 * (C) Copyright sf_Express Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
public class CookieManger implements CookieJar {

    private static final String TAG = "CookieManger";
    private static Context mContext;
    private static CookieStore cookieStore;

    /**
     * Mandatory constructor for the CookieManger
     */
    public CookieManger(Context context) {
        mContext = context;
        if (cookieStore == null) {
            cookieStore = new CookieStore(mContext);
        }
    }

    @Override
    public void saveFromResponse(HttpUrl url, List<Cookie> cookies) {
        if (cookies != null && cookies.size() > 0) {
            for (Cookie item : cookies) {
                cookieStore.add(url, item);
            }
        }
    }

    @Override
    public List<Cookie> loadForRequest(HttpUrl url) {
        List<Cookie> cookies = cookieStore.get(url);
        return cookies;
    }

}

File: MVVMFrame/src/main/java/com/leonyr/mvvm/net/cookie/CookieManger.kt
package com.leonyr.mvvm.net.cookie

import android.content.Context
import com.leonyr.mvvm.net.cookie.CookieManger.Companion.cookieStore

import okhttp3.Cookie
import okhttp3.CookieJar
import okhttp3.HttpUrl

/**
 * ==============================================================
 * Description:
 *
 *
 * Created by 01385127 on 2019.04.29
 * (C) Copyright sf_Express Corporation 2014 All Rights Reserved.
 * ==============================================================
 */
class CookieManger
/**
 * Mandatory constructor for the CookieManger
 */
    (context: Context) : CookieJar {

    init {
        mContext = context
        if (cookieStore == null) {
            cookieStore = CookieStore(mContext)
        }
    }

    override fun saveFromResponse(url: HttpUrl, cookies: List<Cookie>) {
        if (cookies != null && cookies.size > 0) {
            for (item in cookies) {
                cookieStore!!.add(url, item)
            }
        }
    }

    override fun loadForRequest(url: HttpUrl): List<Cookie> {
        return cookieStore!!.get(url)
    }

    companion object {

        private val TAG = "CookieManger"
        private lateinit var mContext: Context
        private var cookieStore: CookieStore? = null
    }

}